### ISSUE TITLE
Use TypeScript instead of JavaScript

### ISSUE THREAD

**Issue by Jaykingamez** [Linnest2020/Vic3-mapdata-editor#3]
It will be pretty annoying to declare types for the various HTML elements, but it has to be done.

Still, if we do so, what bundler should we use rollup or webpack?

**Comment by Linnest2020** [comment]
Yep, Typescript is neccessary for developing, change the repo into Typescript was listed in TODO list, however, it is limitied by two problem

1. I haven't learn how to use Typescript yet
2. Using Typescript may bring some barrier for adding new features, thoght the editor started to get stable

**Comment by Jaykingamez** [comment]
Perhaps you would like to organize it around a framework like React or Angular?

Anyway, the current JS code can be pasted in typescript, the only things to change would be declaring types.

**Comment by Linnest2020** [comment]
https://github.com/Linnest2020/Vic3-mapdata-editor/blob/2855057fd4785de36424eb200602a0e0fa1be658/src/index.ts#L84-L99

You can see I post a ROUGH TypeScript Version on it, and the code above is the interface of `00_states.txt`, however, when it comes with actual usage:
https://github.com/Linnest2020/Vic3-mapdata-editor/blob/2855057fd4785de36424eb200602a0e0fa1be658/src/update/update_states.ts#L32

It would raise error

And then, when I complite with `tsc`, it don't works(warning at `Uncaught ReferenceError: exports is not defined
    at index.js:2:1`)
